### PR TITLE
update package dev documentation and scaffolding

### PR DESCRIPTION
**Pull request by xflr6**
- use requirements.txt to recreate the dev environment
- replace branching model infos with development quickstart
- add development version bump commit to release procedure
- make tox.ini simpler and more widely applicable
- use https for git clone (git@ requires repo write access)

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -2,82 +2,90 @@ Releasing clld
...
 
 - Update translations
-```
-python setup.py extract_messages
-python setup.py update_catalog
-python setup.py compile_catalog
+```shell
+$ python setup.py extract_messages
+$ python setup.py update_catalog
+$ python setup.py compile_catalog
```
Comment (xrotwang) [inline RELEASING.md]: I deliberately left out the shell prompt to make copy&paste of these lines easier.

File: setup.cfg [hunk 2]
```diff
@@ -31,11 +31,17 @@ universal = 1
...
 testpaths = tests
-mock_use_standalone_module = true
 filterwarnings =
     ignore::sqlalchemy.exc.SAWarning
 addopts =
     --appini=tests/test.ini
-    --cov=clld
-    --cov-report term-missing
+    --cov
+    --cov-report=term
```
Comment (xrotwang) [inline setup.cfg]: Since clld does have coverage close to 100%, I think reporting missing lines makes sense.

File: setup.cfg [hunk 3]
```diff
@@ -31,11 +31,17 @@ universal = 1
...
+    --cov-report=html
+mock_use_standalone_module = true
+
+[coverage:run]
+source = clld
 
+[coverage:report]
+show_missing = true
```
Comment (xrotwang) [inline setup.cfg]: Ah ok. So we'll get missing lines after all, right?

File: RELEASING.md (hunk 1 above)
Comment (xflr6) [reply RELEASING.md]: Makes sense, removing.

File: setup.cfg (hunk 2 above)
Comment (xflr6) [reply setup.cfg]: +1 (see below)

File: setup.cfg (hunk 3 above)
Comment (xflr6) [reply setup.cfg]: Yes, we can configure all of coverage in `[coverage:*]` sections of `setup.cfg` like this without needing to look up how to pass it via `pytest-cov`, so I just moved it.

**Comment by xflr6** [comment]
Thanks, `$`s in releasing removed, added `pyramid_debugtoolbar` (hm, wouldn't that be something we only need in concrete apps?) and also `tox` to `dev` extra (travis just does pytest, codecov automatically merges all runs).

**Comment by xrotwang** [comment]
I think it makes sense to aim at installing pyramid and the debugtoolbar at the same time, even if only needed for debugging every once in a while (hence a `dev` requirment).

**Pull request merged**